libs/common/src/main/java/org/opensearch/common/annotation/processor/ApiAnnotationProcessor.java

```diff
@@ -179,6 +179,11 @@ private void process(ExecutableElement executable, ReferenceType ref) {
             return;
         }
 
+        // Skip protobuf generated classes used in public apis
```

Comment:
I think all proto generated classes do have `@Generated` annotation  (if not, it is 100% doable AFAIR) - we could modify `inspectable` method to account for that

Reply:
I can do

```
if (element.getEnclosingElement().getAnnotation(Generated.class) {
  return false;
}
```

in `inspectable` method.

The problem I am facing is the `Generated.class` I am adding in the if condition is from `javax.annotation.processing` while protobuf annotates its code with `javax.annotation`. So the above condition results false. :( 

Any suggestions on how this can be done differently?

Reply:
Fixed this by adding logic in `inspectable`.